Source/WebCore/platform/image-decoders/jpegxl/JPEGXLImageDecoder.cpp

```diff
@@ -437,7 +438,117 @@ LCMSProfilePtr JPEGXLImageDecoder::tryDecodeICCColorProfile()
     return LCMSProfilePtr(cmsOpenProfileFromMem(profileData.data(), profileData.size()));
 }
 
-#endif // USE(LCMS)
+void JPEGXLImageDecoder::maybePerformColorSpaceConversion(void* inputBuffer, void* outputBuffer, unsigned numberOfPixels)
+{
+    if (!m_iccTransform)
+        return;
+
+    cmsDoTransform(m_iccTransform.get(), inputBuffer, outputBuffer, numberOfPixels);
+}
+
+#elif USE(CG)
+
+void JPEGXLImageDecoder::clearColorTransform()
+{
+    m_imageConverter = nullptr;
+}
+
+void JPEGXLImageDecoder::prepareColorTransform()
```

Comment:
Is there any way we can use PixelBufferConverson.h/cpp's functionality to do the work of `prepareColorTransform` and `maybePerformColorSpaceConversion`? `convertImagePixels` is not set up to do a two stage "create the transform" then "apply the transform" conversion, but maybe we can just store the `CGColorSpaceRef` for the embedded ICC profile rather than the `vImageConverterPtr` when we get `JXL_DEC_COLOR_ENCODING`?

Reply:
There is! Updating the patch now.